File: types/company-document.ts
export type DocumentCategory = "policies" | "handbooks" | "templates" | "forms";

export type DocumentFormat = "PDF" | "DOC" | "XLS";

export type DocumentCategoryFilter = "all" | DocumentCategory;

export interface CompanyDocument {
  id: string;
  title: string;
  description: string;
  category: DocumentCategory;
  format: DocumentFormat;
  fileUrl: string;
  fileName: string;
  fileSize: number;
  authorId?: string | null;
  authorName?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  lastAccessedAt?: string | null;
}

export const DOCUMENT_CATEGORIES: {
  id: DocumentCategoryFilter;
  label: string;
}[] = [
  { id: "all", label: "All" },
  { id: "policies", label: "Policies" },
  { id: "handbooks", label: "Handbooks" },
  { id: "templates", label: "Templates" },
  { id: "forms", label: "Forms" },
];

export const CATEGORY_OPTIONS: { id: DocumentCategory; label: string }[] = [
  { id: "policies", label: "Policies" },
  { id: "handbooks", label: "Handbooks" },
  { id: "templates", label: "Templates" },
  { id: "forms", label: "Forms" },
];

File: Backend/app/models/company_document.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field


DocumentCategory = Literal["policies", "handbooks", "templates", "forms"]
DocumentFormat = Literal["PDF", "DOC", "XLS"]


class CompanyDocumentCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = Field(None, max_length=1000)
    category: DocumentCategory
    format: DocumentFormat
    fileUrl: str = Field(..., min_length=1, max_length=2000)
    fileName: str = Field(..., min_length=1, max_length=255)
    fileSize: int = Field(..., ge=0)


class CompanyDocumentUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = Field(None, max_length=1000)
    category: Optional[DocumentCategory] = None
